client/analytics/components/report-chart/index.js

```diff
@@ -180,7 +199,16 @@ ReportChart.propTypes = {
 
 export default compose(
 	withSelect( ( select, props ) => {
-		const { query, endpoint } = props;
+		const { query, endpoint, filters } = props;
+		const chartMode = props.mode || getChartMode( filters, query ) || 'time-comparison';
+
+		if ( 'item-comparison' === chartMode ) {
+			const primaryData = getReportChartData( endpoint, 'primary', query, select );
+			return {
+				primaryData,
+			};
+		}
+
 		const primaryData = getReportChartData( endpoint, 'primary', query, select );
 		const secondaryData = getReportChartData( endpoint, 'secondary', query, select );
```

Comment:
I see the `secondaryData` still being requested. Is that because a variation isn't detected until a products request returns?

Reply:
I can't seem to reproduce this for a variable product. Can you try again and/or let me know what your request looks like?

I added a console.log here, and I only see `secondaryData` when a time comparison chart is happening (a single product that is not variable, or the all products view) -- for comparison charts and the variation view, I don't see the console log.